src/emysql.erl

```diff
@@ -809,3 +807,43 @@ monitor_work(Connection0, Timeout, Args) when is_record(Connection0, emysql_conn
         emysql_conn:reset_connection(emysql_conn_mgr:pools(), Connection, pass),
         exit(mysql_timeout)
     end.
+
+%% @spec execute_transaction(Connection, Fun) -> Result
+%%      Connection = pid()
+%%      Fun = fun()
+
+%%      Result = ok_packet() | result_packet() | error_packet()
+%%
+%% @doc Execute a transaction function
+%%
+%% @private
+%% @end doc: jfjalburquerque july 15
+%%
+execute_transaction(Connection, Fun) ->
+    Result = try
+                 case emysql_conn:execute(Connection, <<"START TRANSACTION;">>, []) of
+                     #ok_packet{} ->
+                         %% execute transaction function
+                         Fun(fun(Query, Params) -> emysql_conn:execute(Connection, Query, Params) end);
+                     MysqlError -> MysqlError
+                 end
+             catch
+                 _:Error ->
+                     %% in case of error in the transaction function
+                     %% we send #error_packet to do rollback
+                     #error_packet{msg = Error}
+             end,
+    case Result of
+        #error_packet{} ->
+            %% in case of #error_packet we execute rollback
+            case emysql_conn:execute(Connection, <<"ROLLBACK;">>, []) of
+                #ok_packet{} -> Result#error_packet{code = 1402, status = <<"XA100">>};
```

Comment:
If the return from `emysql_conn:execute/3` is `#error_packet{}` in this line you are overwriting the `code` and `status` values, is what you want to do? or an error in the transaction always results in an error?